autoscale/src/main/java/com/sequenceiq/periscope/service/UsageReportingService.java

```diff
@@ -0,0 +1,87 @@
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.common.usage.UsageProto;
+import com.sequenceiq.cloudbreak.usage.UsageReporter;
+import com.sequenceiq.periscope.api.model.ScalingStatus;
+import com.sequenceiq.periscope.domain.BaseAlert;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.LoadAlert;
+import com.sequenceiq.periscope.domain.TimeAlert;
+
+@Service
+public class UsageReportingService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UsageReportingService.class);
+
+    @Inject
+    private UsageReporter usageReporter;
+
+    @Async
+    public void reportAutoscalingTriggered(int adjustmentCount, int clusterSize, ScalingStatus scalingStatus, String scalingAction,
+            BaseAlert alert, Cluster cluster) {
+        UsageProto.CDPDatahubAutoscaleTriggered.Builder cdpAutoscaleTriggerDetailsBuilder =
+                UsageProto.CDPDatahubAutoscaleTriggered.newBuilder().setAutoscaleTriggerDetails(
+                        UsageProto.CDPAutoscaleTriggerDetails.newBuilder()
+                                .setAccountId(cluster.getClusterPertain().getTenant())
+                                .setClusterCrn(cluster.getStackCrn())
+                                .setClusterName(cluster.getStackName())
+                                .setAutoscalingAction(scalingAction)
+                                .setAutoscaleTriggerStatus(scalingStatus.name())
+                                .setAutoscalingPolicyDefinition(withAlert(alert))
+                                .setOriginalHostGroupNodeCount(clusterSize)
+                                .setDesiredHostGroupNodeCount(clusterSize + adjustmentCount)
+                                .build());
+
+        usageReporter.cdpDatahubAutoscaleTriggered(cdpAutoscaleTriggerDetailsBuilder.build());
+    }
+
+    @Async
+    public void reportAutoscalingConfigChanged(String userCrn, Cluster cluster) {
+        UsageProto.CDPDatahubAutoscaleConfigChanged.Builder cdpAutoscaleConfigChangedBuilder = UsageProto.CDPDatahubAutoscaleConfigChanged.newBuilder();
+        cdpAutoscaleConfigChangedBuilder.setUserCrn(userCrn);
+        cdpAutoscaleConfigChangedBuilder.setAutoscalingEnabled(cluster.isAutoscalingEnabled());
+        Stream.of(cluster.getLoadAlerts(), cluster.getTimeAlerts())
+                .flatMap(Set::stream)
+                .forEach(alert -> cdpAutoscaleConfigChangedBuilder.addAutoscalingPolicyDefinition(withAlert(alert)));
+
+        usageReporter.cdpDatahubAutoscaleConfigChanged(cdpAutoscaleConfigChangedBuilder.build());
+    }
+
+    public UsageProto.CDPAutoscalePolicyDefinition withAlert(BaseAlert alert) {
+        Map parameters = new HashMap<>();
+        if (alert instanceof TimeAlert) {
+            TimeAlert ta = (TimeAlert) alert;
```

Comment:
Optionally move these into TimeAlert/LoadAlert - so that any changes in parameters, are handled correctly.

Reply:
done